dialog/file.go

```diff
@@ -78,15 +82,17 @@ func (f *fileDialog) makeUI() fyne.CanvasObject {
 		if f.file.save {
 			callback := f.file.callback.(func(fyne.URIWriteCloser, error))
 			name := f.fileName.(*widget.Entry).Text
-			path := filepath.Join(f.dir, name)
+			path := storage.Join(f.dir, name)
 
-			info, err := os.Stat(path)
+			// this assumes the file:// type, which is enforced
+			// when we set `dir`
+			info, err := os.Stat(path.String()[len(path.Scheme())+3:])
 			if os.IsNotExist(err) {
```

Comment:
I think this is the crux of Stat() requirements. We could just check existence when Scheme is file and I think that removes our relience on file URI...

Reply:
I added an `Exists()` function in 78c6984e497079ae24611d2496a576b2237cd726.